packages/matter.js/src/cluster/SwitchCluster.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * @license
+ * Copyright 2022-2023 Project CHIP Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Attribute, OptionalAttribute, Cluster, OptionalEvent, EventPriority } from "./Cluster.js";
+import { TlvUInt8 } from "../tlv/TlvNumber.js";
+import { MatterApplicationClusterSpecificationV1_0 } from "../spec/Specifications.js";
+import { BitFlag } from "../schema/BitmapSchema.js";
+import { TlvField } from "../tlv/TlvObject.js";
+
+/**
+ * Exposes interactions with a switch device for the purpose of using those interactions by other devices
+ *
+ * @see {@link MatterApplicationClusterSpecificationV1_0} § 1.11
+ */
+export const SwitchCluster = Cluster({
+    id: 0x3b,
+    name: "Switch",
+    revision: 1,
+
+    /** @see {@link MatterApplicationClusterSpecificationV1_0} § 1.11.4 */
+    features: {
+        /**  Indicates a switch that maintains its position after being pressed (or turned) */
+        latchingSwitch: BitFlag(0),
+
+        /** Indicates a switch that does not maintain its position after being pressed (or turned)  */
+        momentarySwitch: BitFlag(1),
+
+        /** Indicates a momentary switch that can distinguish and report release events. */
+        momentarySwitchRelease: BitFlag(2),
+
+        /** Indicates a momentary switch that can distinguish and report long presses from short presses. */
+        momentarySwitchLongPress: BitFlag(3),
+
+        /** Idicates a momentary switch that can distinguish and report double press 
+         *  and potentially multiple presses with more events, such as triple press, etc. 
+         */
+        momentarySwitchMultiPress: BitFlag(4),
+    },
+
+    /** @see {@link MatterApplicationClusterSpecificationV1_0} § 1.11.5 */
+    attributes: {
+        /** Indicates the maximum number of positions the switch has with a minimum of 2 */
+        numberOfPositions: Attribute(0, TlvUInt8.bound({ min: 2 }), { default: 2 }),
+
+        /** Indicates the position of the switch with a valid range of zero to NumberOfPositions-1 */
+        currentPosition: Attribute(1, TlvUInt8, { default: 0 }),
+
+        /** Indicates how many consecutive presses can be detected/reported by a multi-press switch */
+        multiPressMax: OptionalAttribute(2, TlvUInt8.bound({ min: 2 }), { default: 2 }),
```

Comment:
Add TODO please with info that mandatory is MSM feature is set ... same please for all evenet below about their constrains

Reply:
Added a bunch of TODOs - kept them on the same line as needs work, but some lines then get very long. Maybe there is a better way to format?

Reply:
Yes I see ... alternative is to move comment to abobe the /** ...*/ line